internal/controllers/machine/machine_controller.go

```diff
@@ -345,21 +345,27 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 
 			conditions.MarkTrue(m, clusterv1.DrainingSucceededCondition)
 			r.recorder.Eventf(m, corev1.EventTypeNormal, "SuccessfulDrainNode", "success draining Machine's node %q", m.Status.NodeRef.Name)
+		}
+
+		// After node draining is completed, and if isNodeVolumeDetachingAllowed returns True, make sure all
+		// volumes are detached before proceeding to delete the Node.
+		if r.isNodeVolumeDetachingAllowed(m) {
+			log.Info("Node volume detaching is allowed for node", "Node", klog.KRef("", m.Status.NodeRef.Name))
```

Comment:
```suggestion
			log.Info("Waiting for node volumes to be detached", "Node", klog.KRef("", m.Status.NodeRef.Name))
```
I would prefer something like this to express what is going on (would be also consistent with the drain log message above)